src/CustomMySQL.js

```diff
@@ -115,7 +116,19 @@ export default class CustomMySQL {
     }
 
     build () {
-        this._cache = Array.from(arguments);
+
+        const args = Array.from(arguments);
+
+        if (args.length === 1 && squel.cls.isSquelBuilder(args[0])) {
```

Comment:
it should be exactly equal to 1 since we still support the old way
```js
// using string
.build('SELECT ?', [1])
.promise()
// using squel builder
const query = squel.select().function('SELECT 1');
...
.build(query)
.promise()
```